app/models/account.rb

```diff
@@ -2,6 +2,12 @@
 
 # Customer organization account
 class Account < ApplicationRecord
+  scope :not_cross_search_tenants, -> {where('settings @> ?', {shared_search: 'false'}.to_json)}
+  has_many :children, class_name: "Account", foreign_key: "parent_id", dependent: :nullify, inverse_of: :parent
+  belongs_to :parent, class_name: "Account", inverse_of: :parent, foreign_key: "parent_id", optional: true
+  store_accessor :settings, :shared_search, :tenant_list
```

Comment:
I don't think you need to store this one boolean field in JSON. maybe just a good old fashioned db field? I see why you might want tenant_list here.

Reply:
I didn't realize that this feature would depend on account settings. it will need a little work to be mergeable but probably not too much